Source/NSubstitute/Exceptions/CouldNotSetReturnException.cs

```diff
@@ -45,6 +45,17 @@ public class CouldNotSetReturnDueToNoLastCallException : CouldNotSetReturnExcept
 #if NET35 || NET4 || NET45
     [Serializable]
 #endif
+    public class CouldNotSetReturnDueToMissingInfoAboutPreviousCallException : CouldNotSetReturnException
+    {
+        public CouldNotSetReturnDueToMissingInfoAboutPreviousCallException() : base("Could not find information about the previous call to return from.") { }
```

Comment:
They only reason why this exception is thrown is incorrect usage of the NSubstitute or bugs, which we are not aware of. At this moment I'm not absolutely sure whether it's even possible to get situation when we have last call router captured, but no pending specification is present. However, I'd assume that it might happen in some exotic scenarios. Therefore this basic warning to use NSubstitute correctly is still relevant and I want to provide it as a main suggestion.

Idea is to distinguish between different situations to improve diagnostics, therefore I don't like idea to throw `CouldNotSetReturnDueToNoLastCallException` for both cases.

If you want to have a different message, clarify which one you would like to have. For now I see that current message is quite correct: there is information about _who_ made the previous call (substitution), but no information _which specs_ were captured.

I could suggest the following
Exception type: `CouldNotSetReturnDueToMissingPendingSpecificationException`
Message: `Could not resolve specification for the previous call.` + Standard suggestions